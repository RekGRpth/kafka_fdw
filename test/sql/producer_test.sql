CREATE FOREIGN TABLE kafka_test_prod (
    part int OPTIONS (partition 'true'),
    offs bigint OPTIONS (offset 'true'),
    some_int int,
    some_text text,
    some_date date,
    some_time timestamp
)
SERVER kafka_server OPTIONS
    (format 'csv', topic 'contrib_regress_prod', batch_size '30', buffer_delay '100');

INSERT INTO kafka_test_prod(some_int, some_text, some_date)
    VALUES
    (1,'foo bar 1','2017-01-01'),
    (2,'foo text 2','2017-01-02'),
    (3,'foo text 3','2017-01-03'),
    (4,'foo text 4','2017-01-04'),
    (5,'foo text 5','2017-01-05'),
    (6,'foo text 6','2017-01-06'),
    (7,'foo bar 7','2017-01-07'),
    (8,'foo text 8','2017-01-08'),
    (9,'foo text 9','2017-01-09'),
    (10,'foo text 10','2017-01-10')

RETURNING *;


SELECT * FROM kafka_test_prod WHERE offs >= 0 and part=0;

INSERT INTO kafka_test_prod(some_int, some_text, some_date, some_time)
SELECT i,
    'It''s some text, that is for number '||i,
    ('2015-01-01'::date + (i || ' minutes')::interval)::date,
    ('2015-01-01'::date + (i || ' minutes')::interval)::timestamp
FROM generate_series(1,1e6::int, 10) i ORDER BY i;


SELECT * FROM kafka_test_prod WHERE offs >= 1000 and part=0 LIMIT 10;