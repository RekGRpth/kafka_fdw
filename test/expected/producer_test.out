CREATE FOREIGN TABLE kafka_test_prod (
    part int OPTIONS (partition 'true'),
    offs bigint OPTIONS (offset 'true'),
    some_int int,
    some_text text,
    some_date date,
    some_time timestamp
)
SERVER kafka_server OPTIONS
    (format 'csv', topic 'contrib_regress_prod', batch_size '30', buffer_delay '100');
INSERT INTO kafka_test_prod(some_int, some_text, some_date)
    VALUES
    (1,'foo bar 1','2017-01-01'),
    (2,'foo text 2','2017-01-02'),
    (3,'foo text 3','2017-01-03'),
    (4,'foo text 4','2017-01-04'),
    (5,'foo text 5','2017-01-05'),
    (6,'foo text 6','2017-01-06'),
    (7,'foo bar 7','2017-01-07'),
    (8,'foo text 8','2017-01-08'),
    (9,'foo text 9','2017-01-09'),
    (10,'foo text 10','2017-01-10')
RETURNING *;
 part | offs | some_int |  some_text  | some_date  | some_time 
------+------+----------+-------------+------------+-----------
      |      |        1 | foo bar 1   | 01-01-2017 | 
      |      |        2 | foo text 2  | 01-02-2017 | 
      |      |        3 | foo text 3  | 01-03-2017 | 
      |      |        4 | foo text 4  | 01-04-2017 | 
      |      |        5 | foo text 5  | 01-05-2017 | 
      |      |        6 | foo text 6  | 01-06-2017 | 
      |      |        7 | foo bar 7   | 01-07-2017 | 
      |      |        8 | foo text 8  | 01-08-2017 | 
      |      |        9 | foo text 9  | 01-09-2017 | 
      |      |       10 | foo text 10 | 01-10-2017 | 
(10 rows)

SELECT * FROM kafka_test_prod WHERE offs >= 0 and part=0;
 part | offs | some_int |  some_text  | some_date  | some_time 
------+------+----------+-------------+------------+-----------
    0 |    0 |        1 | foo bar 1   | 01-01-2017 | 
    0 |    1 |        2 | foo text 2  | 01-02-2017 | 
    0 |    2 |        3 | foo text 3  | 01-03-2017 | 
    0 |    3 |        4 | foo text 4  | 01-04-2017 | 
    0 |    4 |        5 | foo text 5  | 01-05-2017 | 
    0 |    5 |        6 | foo text 6  | 01-06-2017 | 
    0 |    6 |        7 | foo bar 7   | 01-07-2017 | 
    0 |    7 |        8 | foo text 8  | 01-08-2017 | 
    0 |    8 |        9 | foo text 9  | 01-09-2017 | 
    0 |    9 |       10 | foo text 10 | 01-10-2017 | 
(10 rows)

INSERT INTO kafka_test_prod(some_int, some_text, some_date, some_time)
SELECT i,
    'It''s some text, that is for number '||i,
    ('2015-01-01'::date + (i || ' minutes')::interval)::date,
    ('2015-01-01'::date + (i || ' minutes')::interval)::timestamp
FROM generate_series(1,1e6::int, 10) i ORDER BY i;
SELECT * FROM kafka_test_prod WHERE offs >= 1000 and part=0 LIMIT 10;
 part | offs | some_int |                some_text                | some_date  |        some_time         
------+------+----------+-----------------------------------------+------------+--------------------------
    0 | 1000 |     9901 | It's some text, that is for number 9901 | 01-07-2015 | Wed Jan 07 21:01:00 2015
    0 | 1001 |     9911 | It's some text, that is for number 9911 | 01-07-2015 | Wed Jan 07 21:11:00 2015
    0 | 1002 |     9921 | It's some text, that is for number 9921 | 01-07-2015 | Wed Jan 07 21:21:00 2015
    0 | 1003 |     9931 | It's some text, that is for number 9931 | 01-07-2015 | Wed Jan 07 21:31:00 2015
    0 | 1004 |     9941 | It's some text, that is for number 9941 | 01-07-2015 | Wed Jan 07 21:41:00 2015
    0 | 1005 |     9951 | It's some text, that is for number 9951 | 01-07-2015 | Wed Jan 07 21:51:00 2015
    0 | 1006 |     9961 | It's some text, that is for number 9961 | 01-07-2015 | Wed Jan 07 22:01:00 2015
    0 | 1007 |     9971 | It's some text, that is for number 9971 | 01-07-2015 | Wed Jan 07 22:11:00 2015
    0 | 1008 |     9981 | It's some text, that is for number 9981 | 01-07-2015 | Wed Jan 07 22:21:00 2015
    0 | 1009 |     9991 | It's some text, that is for number 9991 | 01-07-2015 | Wed Jan 07 22:31:00 2015
(10 rows)

